lib/body-parser.ts

```diff
@@ -0,0 +1,43 @@
+import * as parse from 'co-body';
+import { Middleware } from '../index';
+import { Context } from 'koa';
+
+declare module 'koa' {
+  interface Request {
+    body: any;
+  }
+}
+
+interface BodyParserOptions {
+  errorCode?: number;
+  errorMessage?: string;
+}
+
+interface FinalOptions extends BodyParserOptions {
+  errorCode: number;
+  errorMessage: string;
+}
+
+const defaultOptions: FinalOptions = {
+  errorCode: 400,
+  errorMessage: 'Unparsable body'
+};
+
+export default function bodyParser(
+  options?: BodyParserOptions
+): Middleware {
+  const finalOptions: FinalOptions = Object.assign(defaultOptions, options);
+
+  return async (
+    ctx: Context,
+    next: () => Promise<any>
+  ): Promise<void> => {
+    try {
+      ctx.request.body = await parse(ctx.req);
+    } catch (er) {
+      ctx.throw(finalOptions.errorCode, finalOptions.errorMessage);
```

Comment:
One thought here — in API we have some middleware that makes sure final error responses are always JSON objects. Do you think that's important? Seems like one less thing to think about on the client side but I don't feel strongly about whether or not this middleware set should include that

Reply:
I certainly like the idea of having a uniform response, even for errors. I'll take a look over there and maybe that's something we can tackle in the near future.